feat(docs): add section about bumping `Cluster` `imageName` using renovate

DerRockWolf opened the pull request:
This regex manager example can be used by users of renovate that want to also automate updates of their CNPG clusters.

Resolves #257

DerRockWolf committed:
feat(docs): add section about bumping `Cluster` `imageName` using renovate

Signed-off-by: RockWolf <git@rockwolf.eu>

acuntex commented:
I've been running this against CNPG tags and hit two practical issues with the current snippet:

1. **Digest is mandatory in the regex**
The pattern requires `@sha256:...`. If a file has only a tag (no digest), Renovate can't extract -> won't open a PR. Making the digest optional lets Renovate both introduce a digest and refresh it later. The README's example currently requires the digest.

2. **File pattern + robustness**
- Consider matching both `.yaml` and `.yml` (and optional `.j2`), not only `.yaml`.
- It's safer if the tag capture stops before `@` (digest) and before inline comments `#`.

Also, it helps to add a `replaceStringTemplate` that writes back the new tag and (optional) digest.

**Proposed doc snippet (drop-in):**

```json5
{
  customManagers: [
    {
      // CloudNativePG Cluster imageName
      customType: "regex",
      description: ["Process custom dependencies"],
      managerFilePatterns: ["/\\.ya?ml(?:\\.j2)?$/"],
      matchStrings: [
        "imageName:\\s*(?<depName>[^:\\s#]+):(?<currentValue>[^@\\s#]+)(?:@(?<currentDigest>sha256:[A-Fa-f0-9]{64}))?"
      ],
      datasourceTemplate: "docker",
      versioningTemplate: "regex:^(?<major>\\d+)\\.(?<minor>\\d+)-(?<patch>\\d+)(?:-(?<compatibility>\\S+))?$",
      replaceStringTemplate: "imageName: {{depName}}:{{newValue}}{{#if newDigest}}@{{newDigest}}{{/if}}"
    }
  ]
}
```

**Why this is safer:**
- Digest is **optional** and strictly validated (`sha256:[A-Fa-f0-9]{64}`), so extraction works before/after pinning.
- `currentValue` stops at `@` and `#`, so the digest and comments aren't swallowed.
- `managerFilePatterns` covers `.yaml`/`.yml` and templated `.j2`.
- `replaceStringTemplate` avoids Renovate writing only the tag and dropping the digest by accident.

DerRockWolf committed:
update matchStrings regex to treat digest as optional

Signed-off-by: RockWolf <git@rockwolf.eu>

DerRockWolf commented:
@acuntex although your answer smells quite a bit AI (correct me if I'm wrong) I'm still going to address a few points.

Firstly, this PR just adds an example renovate manager, so if you need something more specific feel free to adjust it in your config.

The official file extension is `.yaml`. If you have any other extensions in use, like suggested by your template example, please adapt the config for your own setup. `.j2` isn't the only templating "format" out there, so adding just one makes not much sense.

I adapted the regex so that the digest is optional.

By the way, if you don't want to do the entire work on your own at least check that what the LLM spat out is correct. `replaceStringTemplate` doesn't exist anymore / was renamed.

DerRockWolf commented on README.md:
````diff
@@ -243,6 +243,34 @@ cosign verify IMAGE \
+}
+```
+
+Renovate will never change the `compatibility` part of the tag! So bumping from e.g., `trixie` to the next debian release must be done manually.
````
Can you elaborate on that a bit? I thought that at some point after a new Debian release came out you will also provide images for this new version. What else would I as a user need to adapt in order to use the new OS?

DerRockWolf committed:
clarify `compatibility` part

Co-authored-by: Marco Nenciarini <marco.nenciarini@enterprisedb.com>
Signed-off-by: DerRockWolf <50499906+DerRockWolf@users.noreply.github.com>

mnencia committed:
fix autoReplaceStringTemplate to preserve `imageName:` prefix

Renovate replaces the full regex match with the rendered template, so
the original template stripped the `imageName: ghcr.io/...:` prefix
and produced invalid YAML. Including the prefix and `depName` in the
template fixes the replacement while preserving the ability to pin a
digest when one is not already present.

Signed-off-by: Marco Nenciarini <marco.nenciarini@enterprisedb.com>

mnencia committed:
refine Renovate snippet: gate majors, anchor digest, polish prose

Route PostgreSQL major updates through the dependency dashboard for
human approval, and anchor the digest match to exactly 64 hex
characters so malformed digests are no longer captured.

Clarify the trailing prose: the `compatibility` segment pins image
flavour and Debian base (not abstract "compatibility"), and switching
the Debian base is a manual operation because of PostgreSQL
locale-data implications. Note that the example uses JSON5 syntax and
suggest narrowing `managerFilePatterns` when the user repository
contains unrelated YAML.

Signed-off-by: Marco Nenciarini <marco.nenciarini@enterprisedb.com>
